src/Lykke.Job.LiteCoin/Functions/UpdateObservableOperations.cs

```diff
@@ -51,51 +51,68 @@ public class UpdateObservableOperations
-                    
+                return;
+            }
 
-                    var operationCompletedLoggingContext = new
-                    {
-                        unconfirmedTransaction.OperationId,
-                        unconfirmedTransaction.TxHash,
-                        fromAddressUpdatedBalance,
-                        toAddressUpdatedBalance
-                    };
+            var confirmationCount = await _blockChainProvider.GetTxConfirmationCount(unconfirmedTransaction.TxHash);
 
-                    await _operationEventRepository.InsertIfNotExist(OperationEvent.Create(unconfirmedTransaction.OperationId,
-                        OperationEventType.DetectedOnBlockChain, operationCompletedLoggingContext));
+            var isCompleted = confirmationCount >= _confirmationsSettings.MinConfirmationsToDetectOperation;
+            ;
+            if (isCompleted)
+            {
+                //Force update balances
+                var fromAddressUpdatedBalance = await _walletBalanceService.UpdateBalance(operationMeta.FromAddress);
+                var toAddressUpdatedBalance = await _walletBalanceService.UpdateBalance(operationMeta.ToAddress);
 
-                    await _log.WriteInfoAsync(nameof(UpdateBalanceFunctions), nameof(DetectUnconfirmedTransactions),
-                        operationCompletedLoggingContext.ToJson(),
-                        "Operation completed");
 
+                var operationCompletedLoggingContext = new
+                {
+                    unconfirmedTransaction.OperationId,
+                    unconfirmedTransaction.TxHash,
+                    fromAddressUpdatedBalance,
+                    toAddressUpdatedBalance
+                };
 
-                    await _unconfirmedTransactionRepository.DeleteIfExist(unconfirmedTransaction.OperationId);
-                }
+                await _operationEventRepository.InsertIfNotExist(OperationEvent.Create(unconfirmedTransaction.OperationId,
+                    OperationEventType.DetectedOnBlockChain, operationCompletedLoggingContext));
 
-                var status = isCompleted
-                    ? BroadcastStatus.Completed
-                    : BroadcastStatus.InProgress;
+                await _log.WriteInfoAsync(nameof(UpdateBalanceFunctions), nameof(DetectUnconfirmedTransactions),
+                    operationCompletedLoggingContext.ToJson(),
+                    "Operation completed");
 
-                await _observableOperationRepository.InsertOrReplace(ObervableOperation.Create(operationMeta, status,
-                    unconfirmedTransaction.TxHash));
+
+                await _unconfirmedTransactionRepository.DeleteIfExist(unconfirmedTransaction.OperationId);
             }
+
+            var status = isCompleted
+                ? BroadcastStatus.Completed
+                : BroadcastStatus.InProgress;
+
+            var lastBlockHeight = await _blockChainProvider.GetLastBlockHeight();
```

Comment:
I think, block when the transaction was detected as completed, should be guaranteed. This method can be interupted on any service or repository call and repeated later again. Thus more higher block number could be passed to the common part as the block number of the transaction. If between original transaction block and block number, which was passed to the common part, was executed one more transaction, it don't be enrolled to the client.

Add chaos kitty call after every data modification call in this method, to be ready for such cases.